src/engine/MaterializedViews.h

```diff
@@ -0,0 +1,195 @@
+// Copyright 2025 The QLever Authors, in particular:
+//
+// 2025 Christoph Ullinger <ullingec@informatik.uni-freiburg.de>, UFR
+//
+// UFR = University of Freiburg, Chair of Algorithms and Data Structures
+
+#ifndef QLEVER_SRC_ENGINE_MATERIALIZEDVIEWS_H_
+#define QLEVER_SRC_ENGINE_MATERIALIZEDVIEWS_H_
+
+#include "engine/VariableToColumnMap.h"
+#include "index/DeltaTriples.h"
+#include "index/Permutation.h"
+#include "parser/MaterializedViewQuery.h"
+#include "parser/ParsedQuery.h"
+#include "parser/SparqlTriple.h"
+#include "util/HashMap.h"
+
+// Forward declarations
+class QueryExecutionContext;
+class QueryExecutionTree;
+class IndexScan;
+
+// For the future, materialized views save their version. If we change something
+// about the way materialized views are stored, we can break the existing ones
+// cleanly without breaking the entire index format.
+static constexpr size_t MATERIALIZED_VIEWS_VERSION = 1;
+
+// The `MaterializedViewWriter` can be used to write a new materialized view to
+// disk, given an already planned query. The query will be executed lazily and
+// the results will be written to the view.
+class MaterializedViewWriter {
+ private:
+  std::string name_;
+  std::shared_ptr<QueryExecutionTree> qet_;
+  std::shared_ptr<QueryExecutionContext> qec_;
+  ParsedQuery parsedQuery_;
+  ad_utility::MemorySize memoryLimit_;
+  ad_utility::AllocatorWithLimit<Id> allocator_;
```

Comment:
Don't we already have an allocator and therefore also the implicit memory limit inside the `qec`?